File: src/prompts/step2Prompt.js
/**
 * Monta o prompt do Step 2 — Ação.
 *
 * O campo prompt_externo gera um prompt para ser colado em outra IA
 * junto com o PDF do currículo em anexo (sem reproduzir o texto).
 * Inclui exemplo few-shot para calibrar formato e qualidade.
 */
export function buildStep2Prompt({ resume, job, mode, step1Result }) {
  const modeInstructions = {
    geral: 'Gere sugestões gerais de melhoria para o currículo.',
    ats: 'Foque em otimizações para sistemas ATS: adicione palavras-chave, ajuste formatação e estrutura.',
    rewrite: 'Proponha reescritas completas das seções mais fracas, usando linguagem mais impactante e resultados mensuráveis.',
  }

  const instruction = modeInstructions[mode] || modeInstructions.geral

  const jobSection = job
    ? 'DESCRIÇÃO DA VAGA:\n---\n' + job + '\n---\n\n'
    : ''

  const adaptacaoVagaSchema = job
    ? `  "adaptacao_vaga": {
    "match_score": <inteiro de 1 a 10, compatibilidade curriculo x vaga>,
    "palavras_faltando": ["palavra1", "palavra2"],
    "secoes_adaptar": [
      {
        "secao": "<nome da secao>",
        "motivo": "<por que adaptar>",
        "sugestao": "<texto reescrito>"
      }
    ],
    "o_que_destacar": "<string com o que o candidato deve destacar para esta vaga>",
    "o_que_remover": "<string com o que pode ser removido por nao ser relevante para esta vaga>"
  },`
    : ''

  const adaptacaoVagaNote = job
    ? '- "adaptacao_vaga": obrigatorio pois uma vaga foi fornecida'
    : '- "adaptacao_vaga": NAO incluir pois nenhuma vaga foi fornecida'

  const promptExternoRegra =
    '- "prompt_externo": o usuario vai colar este texto em outra IA ' +
    '(ChatGPT, Claude, Gemini) E VAI ANEXAR O PDF DO CURRICULO junto. ' +
    'Por isso o prompt NAO deve reproduzir o texto do curriculo. ' +
    'Ele deve: (1) informar que o curriculo esta em anexo como PDF, ' +
    '(2) listar de forma numerada e detalhada cada melhoria do campo ' +
    '"mudancas" com titulo, descricao e exemplo, ' +
    '(3) pedir que a IA aplique todas as melhorias diretamente no ' +
    'documento preservando o formato original.'

  const fewShotExample =
    'EXEMPLO DE DIAGNOSTICO DO STEP 1:\n' +
    '{"scores":{"impacto":3,"clareza":5,"palavras_chave":4,"geral":4},' +
    '"resumo":"Curriculo vago, sem resultados mensuraveis e verbos fracos."}\n' +
    '\n' +
    'EXEMPLO DE SAIDA ESPERADA DO STEP 2:\n' +
    '{"mudancas":[' +
    '{"tipo":"rewrite","titulo":"Reescrever experiencias com resultados",' +
    '"descricao":"As descricoes atuais usam verbos fracos e nao mostram impacto. Reescreva com verbos de acao e metricas concretas.",' +
    '"exemplo":"Antes: Trabalhou com React (2020-2023)\\nDepois: Desenvolveu 8 features em React/TypeScript para plataforma SaaS com 5k usuarios ativos, reduzindo tempo de carregamento em 35%"},' +
    '{"tipo":"add","titulo":"Adicionar secao de projetos",' +
    '"descricao":"Candidatos sem projetos listados sao descartados por 70% dos recrutadores tech. Adicione 2-3 projetos relevantes.",' +
    '"exemplo":"Projetos: E-commerce API (Node.js + PostgreSQL) — processando 500 pedidos/dia em producao. GitHub: github.com/ana/ecommerce-api"}' +
    '],' +
    '"prompt_externo":"Tenho meu curriculo em anexo (PDF). Por favor, aplique as seguintes melhorias:\\n\\n1. REESCREVER EXPERIENCIAS: substitua descricoes vagas por frases com verbos de acao e metricas. Exemplo — Antes: Trabalhou com React. Depois: Desenvolveu X features em React para Y usuarios, obtendo Z resultado.\\n\\n2. ADICIONAR PROJETOS: inclua uma secao Projetos com 2-3 projetos tecnicos com nome, tecnologias usadas e impacto mensuravel.\\n\\nMantenha o formato e estrutura originais do documento."}\n'

  const body = [
    'Voce e ResumeForge. ' + instruction,
    '',
    fewShotExample,
    'Agora gere as sugestoes para o curriculo abaixo. Retorne APENAS o JSON:',
    '',
    'DIAGNOSTICO DO STEP 1:',
    JSON.stringify(step1Result, null, 2),
    '',
    'CURRICULO:',
    '---',
    resume,
    '---',
    '',
    jobSection + 'Retorne SOMENTE este JSON:',
    '{',
    '  "mudancas": [',
    '    {',
    '      "tipo": "<add | edit | remove | rewrite>",',
    '      "titulo": "<titulo curto>",',
    '      "descricao": "<o que mudar e por que>",',
    '      "exemplo": "<antes -> depois concreto>"',
    '    }',
    '  ],',
    adaptacaoVagaSchema,
    '  "prompt_externo": "<prompt para colar em outra IA junto com o PDF em anexo>"',
    '}',
    '',
    'Regras:',
    '- "mudancas": entre 5 e 10 itens',
    '- Tipos: add | edit | remove | rewrite',
    '- Exemplos devem ser concretos com antes e depois real',
    '- LIMITE DE ESPACO: o curriculo tem no maximo 2 paginas. Sugestoes do tipo "add" so sao validas se houver espaco real (ex: remover algo para abrir espaco, ou condensar texto existente). Nunca sugira adicionar secoes inteiras sem considerar o espaco disponivel.',
    '- Prefira sugestoes de rewrite e edit que melhorem o impacto sem aumentar o tamanho, ou sugestoes de remove que liberem espaco para algo mais valioso.',
    promptExternoRegra,
    adaptacaoVagaNote,
  ].join('\n')

  return body
}

File: backend/prompts/step2_prompt.py
import json

def build_step2_prompt(resume: str, job: str, mode: str, step1_result: dict) -> str:
    job_section = ""
    if job and job.strip():
        job_section = f"""
DESCRIÇÃO DA VAGA:
{job}

Para adaptação_vaga, identifique palavras-chave da vaga ausentes no currículo e ajustes específicos.
Se houver PDF da vaga anexado (prompt_externo), priorize as informações do PDF.
"""
    else:
        job_section = "\nNenhuma vaga específica fornecida. Pule o campo adaptacao_vaga (retorne null).\n"

    mode_instructions = {
        "geral": "Sugestões gerais para melhorar o currículo.",
        "tech": "Foque em stack técnica, projetos, métricas de engenharia.",
        "gestao": "Foque em liderança, resultados de negócio, gestão de equipes.",
    }
    mode_text = mode_instructions.get(mode, mode_instructions["geral"])

    problems = step1_result.get("problemas", [])
    scores = step1_result.get("scores", {})

    return f"""Você é um especialista em currículos. Com base na análise já feita, gere sugestões concretas e acionáveis de melhoria.

MODO: {mode_text}

ANÁLISE JÁ FEITA:
- Nota geral: {scores.get('geral', '?')}/10
- Problemas identificados: {json.dumps(problems, ensure_ascii=False)}
{job_section}
CURRÍCULO ATUAL:
{resume}

REGRAS:
1. Cada mudança deve mostrar o texto original E o texto melhorado (quando aplicável).
2. Seja específico — não diga "adicione métricas", mostre como ficaria a frase com a métrica.
3. Limite a 2 páginas — não sugira adicionar seções desnecessárias. Se já está cheio, sugira substituições.
4. Priorize mudanças de alto impacto (as que mais melhoram a nota).
5. No máximo 6 mudanças.

EXEMPLO DE MUDANÇA BEM FORMATADA:
{{
  "secao": "Experiência — TechCorp",
  "original": "Trabalhei no desenvolvimento de sistemas.",
  "sugerido": "Desenvolvi APIs REST em Python/FastAPI, reduzindo latência média em 40% (de 500ms para 300ms).",
  "motivo": "Especifica tecnologia, adiciona métrica quantitativa de impacto."
}}

Retorne SOMENTE o JSON, sem markdown:
{{
  "mudancas": [
    {{
      "secao": "<nome da seção>",
      "original": "<texto atual ou null se novo>",
      "sugerido": "<texto melhorado>",
      "motivo": "<por que essa mudança melhora o currículo>"
    }}
  ],
  "adaptacao_vaga": {{
    "palavras_chave_ausentes": ["<keyword 1>", ...],
    "ajustes_recomendados": ["<ajuste específico>", ...]
  }}
}}"""
